sm/download/src/download.rs

```diff
@@ -142,6 +181,30 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn downloader_download_with_content_length() -> anyhow::Result<()> {
+        let tmpstats = statvfs::statvfs("/tmp")?;
+        let usable_disk_space = tmpstats.blocks_free() * tmpstats.block_size();
+        let _mock1 = mock("GET", "/some_file.txt")
+            .with_header("content-length", &(usable_disk_space.to_string()))
```

Comment:
I might be good to use the `test-case` crate and to add a content size parameter to this test in order to test something more realistic.

Reply:
I might also be good to add positive tests (with small files but large enough to exercise the copy loop) as well as test without a content length.